src/DRICH_geo.cpp

```diff
@@ -491,52 +518,53 @@ static Ref_t createDetector(Detector& desc, xml::Handle_t handle, SensitiveDetec
 
     // thetaGen loop: iterate less than "0.5 circumference / sensor size" times
     double nTheta = M_PI * sensorSphRadius / (sensorSide + sensorGap);
-    for (int t = 0; t < (int)(nTheta + 0.5); t++) {
-      double thetaGen = t / ((double)nTheta) * M_PI;
-
-      // phiGen loop: iterate less than "circumference at this latitude / sensor size" times
-      double nPhi = 2 * M_PI * sensorSphRadius * std::sin(thetaGen) / (sensorSide + sensorGap);
-      for (int p = 0; p < (int)(nPhi + 0.5); p++) {
-        double phiGen = p / ((double)nPhi) * 2 * M_PI - M_PI; // shift to [-pi,pi]
-
-        // determine global phi and theta
-        // - convert {radius,thetaGen,phiGen} -> {xGen,yGen,zGen}
-        double xGen = sensorSphRadius * std::sin(thetaGen) * std::cos(phiGen);
-        double yGen = sensorSphRadius * std::sin(thetaGen) * std::sin(phiGen);
-        double zGen = sensorSphRadius * std::cos(thetaGen);
-        // - convert {xGen,yGen,zGen} -> global {x,y,z} via rotation
-        double x = zGen;
-        double y = xGen;
-        double z = yGen;
-        // - convert global {x,y,z} -> global {phi,theta}
-        // double phi   = std::atan2(y, x);
-        // double theta = std::acos(z / sensorSphRadius);
-
-        // shift global coordinates so we can apply spherical patch cuts
-        double zCheck   = z + sensorSphCenterZ;
-        double xCheck   = x + sensorSphCenterX;
-        double yCheck   = y;
-        double rCheck   = std::hypot(xCheck, yCheck);
-        double phiCheck = std::atan2(yCheck, xCheck);
-
-        // patch cut
-        bool patchCut = std::fabs(phiCheck) < sensorSphPatchPhiw && zCheck > sensorSphPatchZmin &&
-                        rCheck > sensorSphPatchRmin && rCheck < sensorSphPatchRmax;
-        if (debugSensors)
-          patchCut = std::fabs(phiCheck) < sensorSphPatchPhiw;
-        if (patchCut) {
-
-          // append sensor position to centroid calculation
-          // if (isec == 0) {
-          //   sensorCentroidX += xCheck;
-          //   sensorCentroidZ += zCheck;
-          //   sensorCount++;
-          // };
-
-          // placement (note: transformations are in reverse order)
-          // - transformations operate on global coordinates; the corresponding
-          //   generator coordinates are provided in the comments
-          auto sensorPlacement =
+    if(debugOpticsMode != 4){
+      for (int t = 0; t < (int)(nTheta + 0.5); t++) {
+	double thetaGen = t / ((double)nTheta) * M_PI;
+
```

Comment:
Avoid C-style casts. In particular, integers are upcast to double when needed. It may be that `t` does not have to be an integer here; `for` loops work fine with non-integer types too.